feat: 과제 공고 등록  기능 구현

d4eh0 opened the pull request.

d4eh0 committed:
feat: Assignment 엔티티 초안 생성

d4eh0 committed:
feat: Assignment 엔티티 생성

d4eh0 committed:
feat: Assignment 요청/응답 DTO 생성

d4eh0 committed:
chore: project브랜치 변경사항 반영

d4eh0 committed:
refactor: Assignment 엔티티 및 creationRequestDto 수정

- 만들었던 RecruitmentParticipants 필드를 제거하고 ParticipantInfo로 교체

- Assignment 엔티티 grades를 List → Set으로 변경반영

d4eh0 committed:
feat: 과제 공고 등록 기능 구현
- AssignmentCreationRequestDto : final 필드 누락 보완
- BaseRecruitmentRequestDto : @FutureOrPresent 검증 추가
- Controller, Service, Repository 생성

d4eh0 committed:
Merge remote-tracking branch 'origin/develop' into feat/BE-27-과제-공고-등록-API

d4eh0 committed:
refactor: Assignment 엔티티 및 생성 로직 개선
- Assignment 엔티티 @Table(name = assignments)로 명시
- AssignmentController에 인증 체크(@PreAuthorize) 추가
- grades 변환 로직 for문 → Stream API로 변경

d4eh0 committed:
feat: Assignment 엔티티 생성 테스트 코드 추가

d4eh0 committed:
Merge remote-tracking branch 'origin/develop' into feat/BE-27-과제-공고-등록-API

ming920 commented on src/main/java/com/wagglex2/waggle/domain/assignment/service/serviceImpl/AssignmentServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
```
@Transactional(readOnly = true)로 바꾸시고 밑에 메서드에 @Transactional로 하는게 좋을 거 같아요

ojm6135 commented on src/main/java/com/wagglex2/waggle/domain/assignment/dto/request/AssignmentCreationRequestDto.java:
```diff
@@ -0,0 +1,68 @@
+
+    @NotNull(message = "모집 인원이 누락되었습니다.")
+    @Valid
+    private final ParticipantInfo participants;
```
과제는 포지션 구분이 없고, 등록 때는 모집 인원만 필요하니까
maxParticipants로 바로 받아도 될 거 같아요.

ojm6135 commented on src/main/java/com/wagglex2/waggle/domain/assignment/service/serviceImpl/AssignmentServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+    private final UserService userService;
+
+    @Override
+    public void createAssignment(AssignmentCreationRequestDto requestDto, Long userId) {
```
오늘 회의에서 공고 등록 후 id를 응답해주기로 했어요.

d4eh0 committed:
refactor: 과제 공고 등록 기능 수정

- 트랜잭션 범위 조정
- 공고 등록 시 생성된 id 리턴
- creationDTO maxParticipants 받게 변경
- AssignmentRepositoryTest 작성
